src/Npgsql/NpgsqlDataSourceBuilder.cs

```diff
@@ -153,6 +155,26 @@ public NpgsqlDataSourceBuilder UseClientCertificatesCallback(Action<X509Certific
         return this;
     }
 
+    /// <summary>
+    /// Sets the <see cref="X509Certificate2" /> that will be used validate SSL certificate, received from the server.
+    /// </summary>
+    /// <param name="caCertificate">The CA certificate.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlDataSourceBuilder UseCACertificate(X509Certificate2? caCertificate)
+        => UseCACertificateCallback(() => caCertificate);
+
+    /// <summary>
+    /// Specifies a callback that will be used to validate SSL certificate, received from the server.
+    /// </summary>
+    /// <param name="caCertificateCallback">The callback to get CA certificate.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlDataSourceBuilder UseCACertificateCallback(Func<X509Certificate2?>? caCertificateCallback)
```

Comment:
Any reason the X509Certificate2 Func parameter needs to be nullable?

Reply:
Not really, just simplified things a little.